resources/migrations/001_update_migrations.yaml

```
@@ -6494,6 +6502,193 @@ databaseChangeLog:
+        - sql:
+            dbms: postgresql
+            sql: >-
+              DROP INDEX IF EXISTS idx_uniq_field_table_id_parent_id_name_2col;
+              CREATE UNIQUE INDEX idx_uniq_field_table_id_parent_id_name_2col
+                ON "metabase_field" ("table_id", "name")
+                WHERE "parent_id" IS NULL AND "is_defective_duplicate" = FALSE;
+      rollback: # No rollback since we want to allow downgrades to 48 to succeed
+
+  # The next two changesets replace ON DELETE CASCADE with ON DELETE RESTRICT on fk_field_parent_ref_field_id.
+  # We need to do this for MySQL because it doesn't support ON DELETE CASCADE in a stored generated column, and we
+  # want to use parent_id in the `unique_field_helper` generated column. Cascading deletes are handled in the
+  # application instead.
+  - changeSet:
+      id: v49.2024-06-27T00:00:05
+      author: calherries
+      comment: Drop fk_field_parent_ref_field_id constraint with ON DELETE CASCADE
+      changes:
+        - dropForeignKeyConstraint:
+            baseTableName: metabase_field
+            constraintName: fk_field_parent_ref_field_id
+      rollback:
+        - addForeignKeyConstraint:
+            baseTableName: metabase_field
+            baseColumnNames: parent_id
+            referencedTableName: metabase_field
+            referencedColumnNames: id
+            constraintName: fk_field_parent_ref_field_id
+            onDelete: CASCADE
+
+  - changeSet:
+      id: v49.2024-06-27T00:00:06
+      author: calherries
+      comment: Add fk_field_parent_ref_field_id constraint without ON DELETE RESTRICT
+      changes:
+        - addForeignKeyConstraint:
+            baseTableName: metabase_field
+            baseColumnNames: parent_id
+            referencedTableName: metabase_field
+            referencedColumnNames: id
+            constraintName: fk_field_parent_ref_field_id
+      rollback:
+        - dropForeignKeyConstraint:
+            baseTableName: metabase_field
+            constraintName: fk_field_parent_ref_field_id
+
+  # The following two changesets add a new column `unique_field_helper` to `metabase_field` and a unique constraint
+  # `idx_unique_field` for H2 and MySQL.
+  # The combination of `unique_field_helper` and `idx_unique_field` achieves two things:
+  # 1. It enforces the conditional constraints:
+  #   - if parent_id IS NULL, then (table_id, name) is unique. (Postgres' idx_uniq_field_table_id_parent_id_name_col2)
+  #   - if parent_id IS NOT NULL, then (table_id, name, parent_id) is unique. (Postgres' idx_uniq_field_table_id_parent_id_name)
+  # 2. It only enforces the constraints when is_defective_duplicate = FALSE
+  # `unique_field_helper` is only used in H2 and MySQL but it is added in Postgres so all DBs have the same set of colums.
+  - changeSet:
+      id: v49.2024-06-27T00:00:07
+      author: calherries
+      comment: Add unique_field_helper column to metabase_field
+      changes:
+        - sql:
+            dbms: postgres
```

```suggestion
            dbms: postgresql
```

man, I bet other people will have this mistake too, probably need a linter for this.

---

wow, appreciate your sharp eyes. thanks